cpp/skyweaver/detail/MultiFileWriter.cu

```diff
@@ -241,23 +242,73 @@ bool MultiFileWriter<VectorType>::operator()(VectorType const& stream_data,
                                              std::size_t stream_idx)
 {
     if(!has_stream(stream_idx)) {
+        if (_config.get_wait_config().is_enabled) {
```

Comment:
This is a common block of logic an should be factored out.

Reply:
That is true. Is there a good linter to find such flaws?

Reply:
Apparently [simian](https://simian.quandarypeak.com/docs/) but I've never used it.